src/deepsparse/pipeline.py

```diff
@@ -220,8 +228,30 @@ def __call__(self, *args, **kwargs) -> BaseModel:
                 f"Outputs of {self.__class__} must be instances of "
                 f"{self.output_schema} found output of type {type(pipeline_outputs)}"
             )
+        timer.stop("post_process")
+        timer.stop("total_inference")
+
+        if not monitoring:
+            return pipeline_outputs
+
+        else:
+            data = None
+            timing = timer.build()
+            return pipeline_outputs, timing, data
 
-        return pipeline_outputs
+    def run_with_monitoring(self, *args, **kwargs) -> Tuple[BaseModel, BaseModel, Any]:
```

Comment:
Is there a case when we wouldn't want to be monitoring? wondering if we can just always monitor and delete this extra function

Reply:
Agree with you there - then I vote for the idea in my next comment down about additional method for retrieiving timings instead of the monitoring flag 😁